Dev

darenwong opened the pull request:
This version of pull request have PostgreSQL and Node Server hosted on Heroku. 

The endpoint for heroku is https://floating-bridge-server.herokuapp.com

darenwong committed:
Chatbox is now a toggleable pop-up

Commit:
Add README for dev

darenwong committed:
Added animation to the board in App module and Board module

darenwong committed:
Merge branch 'readme' of https://github.com/darenwong/React-online-bridge-app into feature-chatbox

darenwong committed:
Made cards on board stack in right order and made bid cards animation prettier on hover

darenwong committed:
Animated Chatbox notification badge with CSS

Chatbox notification badge dissapears if there are no new messages. When there are new messages, the badge appears and translate vertically in a loop. When chatbox is opened, notification number is set to 0 and the badge dissapears

darenwong committed:
Added display to indicate trump broken status and partner revealed status

Added 2 room states, partnerRevealed and partner. When server receives a card played by client that matches the partner card ID, the partner container turns red and the partner playing position is revealed. Similarly, when client receive state update that trump is broken, the contract container turns red.

darenwong committed:
Server now sends message to inform when a new user joins a room. AI also check if there are valid cards in hand before playing
Added check in AI class to check if there are valid cards in hand to play. If none, return nothing

darenwong committed:
Commit to push db to heroku

darenwong committed:
Fixed dependency: Added Joi as dependency

darenwong committed:
Committed with changes to db credentials for heroku deployment

darenwong committed:
set SSL rejectUnauthorized to false so that node can query db

darenwong committed:
Updated server endpoint so that client connects to heroku instead of localhost

darenwong committed:
Clients can now view the last trick via a toggleable button

Feature and button resides within board.js

darenwong committed:
Fixed bug on last trick viewing feature

Previously, both chatbox and lasttrick overlaps when both are toggled. Now, toggling one will close the other

darenwong committed:
Added missing dependencies for web side as following:

react-router-dom
react-scroll
typewriter-effect

darenwong committed:
Board now highlight who the winner is by dimming the rest

This feature is applied to both current round and last trick

darenwong committed:
Fixed Bug: Added back update trump broken status script that was previously commented

darenwong committed:
Fixed Bug: Added back update trump broken status script that was previously commented

darenwong committed:
Merge pull request #7 from darenwong/feature-view-last-trick

Incorporating new feature (view last trick played) into dev. This feature consists of:

1. Allow user to view what was the last trick that was played.
2. Highlights the card that won the current and previous round.
3. When last trick button is toggled, chat button is untoggled if it was toggled, and vice versa.

BeardedBread commented:
I performed a clean clone and run the instructions. The client threw an error:
/src/components/board.jsx
Module not found: Can't resolve 'react-icons/ai' in '<WORKING_DIR>/web/src/components'

It seems to be missing a module named react-icons. Could you fix this?

After that, I think you should just merge it. I've tested it after installing the missing dependencies. I can play a full game without any issues as a guest. So, that's good enough.

darenwong committed:
Randomised bid starting position

Changes were made to room.js in Server folder to set initial room state for turns to a random integer between 0 to 3 (inclusive)

darenwong committed:
Added sound effects

Sound effects is ran whenever turn has changed. This means every bid and card played will run the sound effects. Sound effect also added to when mouse hovers over player's hand

darenwong committed:
Tidied up code by removing previously commented code

darenwong committed:
Added side bar (built with material ui) with following features:

View Game info such as who won the bid, partner's card etc
View bid log
New settings feature allows background to be changed

darenwong committed:
Restyled app and  board components to make it more responsive

Still work in progress.
Role button and cards are mostly responsive now.
TO DO: make bid, selectPartner responsive.

darenwong committed:
Further improved responsiveness of website

Still work in progress

darenwong committed:
Improved responsiveness by putting chat, last trick and menu buttons inside an AppBar

Still work in progress
Bugs to fix:
Hand cards offset overshoots
chat badge number sometimes dont return to zero when chat is active

darenwong committed:
Put selectRole button in a dialog instead of dropdown and fixed following bugs:

Fixed card left margin offset.
Fixed chat notification number reset when chatbox is active.

darenwong committed:
Set up Carousel to display gameplay screenshots in home page, converted loginPage into Dialogs using Material UI

darenwong committed:
Cleaned up unused imported modules and logs for production

darenwong committed:
Merge pull request #8 from darenwong/feature-sound

This production version is mainly on enhancing responsiveness and improving components display using Material UI

darenwong committed:
Update README.md

darenwong committed:
Updated web page header icon and title

darenwong committed:
Added loading transition between request and response to and from server

darenwong committed:
Secret states are now preserved, tidied up updateState function

Secret states like other player's cards and partner identity are contained within server

darenwong committed:
Added Error Dialog to alert user

darenwong committed:
Removed script used during development

darenwong committed:
Server now auto deletes empty room

darenwong committed:
Fixed displaying error message in Login Page

darenwong committed:
Removed server feature to auto delete empty room

darenwong committed:
Merge branch 'dev' into feature-loading

darenwong committed:
Merge pull request #9 from darenwong/feature-loading

Added loading animation while waiting for server response, and preserve secret states
